src/braket/ir/jaqcd/program_v1.py

```diff
@@ -157,6 +161,18 @@ class Program(BraketSchemaBase):
 
     _PROGRAM_HEADER = BraketSchemaHeader(name="braket.ir.jaqcd.program", version="1")
     braketSchemaHeader: BraketSchemaHeader = Field(default=_PROGRAM_HEADER, const=_PROGRAM_HEADER)
-    instructions: List[GateInstructions]
+    instructions: List[Any]
     results: Optional[List[Results]]
-    basis_rotation_instructions: Optional[List[GateInstructions]]
+    basis_rotation_instructions: Optional[List[Any]]
+
+    @validator("instructions", "basis_rotation_instructions", each_item=True, pre=True)
+    def validate_instructions(cls, value, field):
+        """
+        Pydantic uses the validation subsystem to create objects. We use this custom validator for
+        2 reasons:
+        1. Implement O(1) deserialization
+        2. Validate that the input instructions are supported
+        """
+        if value is not None and "type" in value and value["type"] not in _valid_gates:
+            raise ValueError(f"Invalid gate specified: {value} for field: {field}")
```

Comment:
Maybe add cases for `value is None` and `"type" not in value`? Might be overly paranoid, but it would at least be nice to have some clear information if these pathological cases somehow occur. The default `KeyError: <type>` and `NoneType` errors can be confusing.